sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -125,6 +263,32 @@ def gen_df_rolling_method_impl(method_name, self, args=None, kws=None):
     return _impl
 
 
+@sdc_overload_method(DataFrameRollingType, 'corr')
+def sdc_pandas_dataframe_rolling_corr(self, other=None, pairwise=None):
+
+    ty_checker = TypeChecker('Method rolling.corr().')
+    ty_checker.check(self, DataFrameRollingType)
+
+    accepted_other = (bool, Omitted, NoneType, DataFrameType, SeriesType)
+    if not isinstance(other, accepted_other) and other is not None:
+        ty_checker.raise_exc(other, 'DataFrame, Series', 'other')
+
+    accepted_pairwise = (bool, Boolean, Omitted, NoneType)
+    if not isinstance(pairwise, accepted_pairwise) and pairwise is not None:
+        ty_checker.raise_exc(pairwise, 'bool', 'pairwise')
+
+    nan_other = isinstance(other, (Omitted, NoneType)) or other is None
+    kws = {'other': 'None', 'pairwise': 'None'}
+
+    if nan_other:
+        return gen_df_rolling_method_without_other_impl('corr', self, kws=kws)
+
+    if isinstance(other, DataFrameType):
+        return gen_df_rolling_method_with_other_df_impl('corr', self, other, kws=kws)
+
+    return gen_df_rolling_method_impl('corr', self, kws=kws)
```

Comment:
@densmirn I think this can be a bit improved by always passing the other type into the real generation function (e.g. df_rolling_method_codegen) and inside it selecting a branch based on the other type (which can be None, Omitted, DataFrame and Series). This way we can join the functions: df_rolling_method_without_other_codegen and df_rolling_method_codegen, which are almost the same into one, e.g.

```
def df_rolling_method_codegen(method_name, self, other, args=None, kws=None):
    args = args or []
    kwargs = kws or {}

    impl_params = ['self'] + args + params2list(kwargs)
    impl_params_as_str = ', '.join(impl_params)

    impl_name = f'_df_rolling_{method_name}_impl'
    func_lines = [f'def {impl_name}({impl_params_as_str}):']

    if not other:
        if 'pairwise' in kwargs:
            func_lines += [
                '  if pairwise is None:',
                '    _pairwise = True',
                '  else:',
                '    _pairwise = pairwise',
                '  if _pairwise:',
                '    raise ValueError("Method rolling.corr(). The object pairwise\\n expected: False")'
            ]

        method_params = args + ['{}={}'.format(k, k) for k in kwargs if k != 'other']
    else:
        method_params = args + ['{}={}'.format(k, k) for k in kwargs]
    func_lines += df_rolling_method_tail_codegen(method_params, self.data.columns, method_name)
    func_text = '\n'.join(func_lines)

    global_vars = {'pandas': pandas, 'get_dataframe_data': get_dataframe_data}

    return func_text, global_vars
```
What do you think?

Reply:
Done!